openmdao/components/eq_constraint_comp.py

```diff
@@ -130,9 +130,10 @@ def compute(self, inputs, outputs):
             # scale factor that normalizes by the rhs, except near 0
             if options['normalize']:
                 # Indices where the rhs is near zero or not near zero
-                idxs_nz = np.where(cs_safe.abs(rhs) < 2)[0]
-                idxs_nnz = np.where(cs_safe.abs(rhs) >= 2)[0]
+                idxs_nz = np.where(cs_safe.abs(rhs) < 2)
+                idxs_nnz = np.where(cs_safe.abs(rhs) >= 2)
 
+                self._scale_factor = np.ones((rhs.shape))
```

Comment:
The missing part of this story is that we are saving `_scale_factor ` and `_dscale_drhs` in the class, but we are only saving the value from the last variable in the for loop. Choices were: 1) save in dictionary keyed by varname, or 2) don't save at all and remove it everywhere except where it is calculated.

Your job on this story is to evaluate whether we really *need* to save it, and make the call between 1 and 2.  I think Rob has already indicated that we don't need to query it after the model runs, so it's just a matter of whether the values are re-used internally.